### ISSUE TITLE
Incosistent argument in the velocity analysis example

### ISSUE THREAD

**Issue by subercui** [dynverse/dyngen_manuscript#3]
Hi, when using the [usecase_rna_velocity](https://github.com/dynverse/dyngen_manuscript/blob/8370bf5ce8a4c9e78967122202ccfee56fd516f1/analysis/usecase_rna_velocity/1_datasets.R#L19)
```
pwalk(design_datasets, function(id, seed, backbone) {
  if (!file.exists(exp$dataset_file(id))) {
    cat("## Generating ", id, "\n", sep = "")
    set.seed(seed)
    backbone <- dyngen::list_backbones()[[backbone]]()
    model <-
      initialise_model(
        id = id,
        num_tfs = 20,
        num_targets = 50,
        num_hks = 15,
        backbone = backbone,
        num_cells = 1000,
        simulation_params = simulation_default(
          compute_propensity_ratios = TRUE
        ),
        num_cores = 4,
        download_cache_dir = "~/.cache/dyngen",
        verbose = TRUE
      )
    generate_dataset(
      model,
      output_dir = exp$dataset_folder(id),
      make_plots = TRUE,
      store_propensity_ratios = TRUE
    )
  }

  gc()
})
```
I got an error: `Error in initialise_model(id = id, num_tfs = 20, num_targets = 50, num_hks = 15,  : 
  unused argument (id = id)`
Seems the interface is incosistent? Do you have any idea?

**Comment by rcannood** [comment]
usecase_rna_velocity uses the devel version of dyngen. I'll merge the devel version of dyngen into master. 
Could you reinstall dyngen and try again? :)

**Comment by subercui** [comment]
Thank you for the prompt update! It solves the argument inconsistency, but the dynplot_dimred is not found [here](https://github.com/dynverse/dyngen_manuscript/blob/8370bf5ce8a4c9e78967122202ccfee56fd516f1/analysis/usecase_rna_velocity/1_datasets.R#L63)

`
dynplot_dimred(dataset) + 
      geom_cell_point()                                                                                                                                                                                 
Error in dynplot_dimred(dataset) : 
  could not find function "dynplot_dimred"
`
Does the dynplot2 need to update as well?

**Comment by rcannood** [comment]
Ahh yes, you need the latest dynplot2. According to @zouter, this branch is still very much WIP, so let us know if you encounter any issues there.

Please run
```r
devtools::install_github("dynverse/dynplot2@devel")
```

**Comment by subercui** [comment]
Encountering some weird error showing `repo name 'scvelo'`. I guess maybe I can try to install manually but here is the error message:
```
devtools::install_github("dynverse/dynplot2@devel")  
Downloading GitHub repo dynverse/dynplot2@devel
Error: Failed to install 'dynplot2' from GitHub:
  HTTP error 404.
  No commit found for the ref devel

  Did you spell the repo owner (`dynverse`) and repo name (`scvelo`) correctly?
  - If spelling is correct, check that you have the required permissions to access the repo.
```

**Comment by subercui** [comment]
Also, when running `2_velocity.R`, this line [scvelo::get_velocity](https://github.com/dynverse/dyngen_manuscript/blob/master/analysis/usecase_rna_velocity/2_velocity.R#L60) raises error `Error in loadNamespace(name) : there is no package called ‘scvelo’`, although I can successfully import using `reticulate::import("scvelo")`.

I think some misconfiguration related to `scvelo`.

**Comment by rcannood** [comment]
> Encountering some weird error showing repo name 'scvelo'. I guess maybe I can try to install manually but here is the error message:

I see you managed to get scvelo (the R package) installed manually. In the meantime, I also fixed the underlying problem.

> Also, when running 2_velocity.R, this line scvelo::get_velocity raises error Error in loadNamespace(name) : there is no package called ‘scvelo’, although I can successfully import using reticulate::import("scvelo").

This is because by default, reticulate uses r-miniconda. There are two approaches to solving this:

#### 1. Install scvelo in whatever environment reticulate is using
By running `scvelo::install_scvelo()` or `reticulate::py_install(packages = "git+https://github.com/theislab/scvelo.git", pip = TRUE)`.

#### 2. Specify which python to use
Either specify the conda environment you use (`reticulate::use_condaenv("yourcondaenv")`), or let reticulate know that you wish to use the default python (`reticulate::use_python("/usr/bin/python3", required = TRUE)`).

**Comment by subercui** [comment]
Yes, the problem is solved. The `dynplot2@devel` is installed successfully and then the `scvelo` package works well.
Thanks again!
